src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -18,97 +26,80 @@ class TextColorToGenerator : IIncrementalGenerator
-			// We just generate a method if the Control is a new implementation of ITextStyle and IAnimatable
-			var doesContainSymbolBaseType = mauiTextStyleImplementors.ContainsSymbolBaseType(declarationSymbol);
+				var iTextStyleInterfaceSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+				var iAnimatableInterfaceSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
 
-			if (!doesContainSymbolBaseType
-				&& declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
-				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
-			{
-				if (declarationSymbol.ContainingNamespace.IsGlobalNamespace)
+				if (iTextStyleInterfaceSymbol is null || iAnimatableInterfaceSymbol is null)
 				{
-					var diag = Diagnostic.Create(TextColorToDiagnostic.GlobalNamespace, Location.None, declarationSymbol.Name);
-					context.ReportDiagnostic(diag);
-					continue;
+					throw new Exception("There's no .NET MAUI referenced in the project.");
 				}
 
-				var nameSpace = declarationSymbol.ContainingNamespace.ToDisplayString();
+				var mauiAssembly = compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == mauiControlsAssembly);
+				var symbols = GetMauiInterfaceImplementors(mauiAssembly, iAnimatableInterfaceSymbol, iTextStyleInterfaceSymbol).Where(x => x is not null);
 
-				var accessModifier = GetClassAccessModifier(declarationSymbol);
+				return symbols;
+			});
 
-				if (accessModifier == string.Empty)
-				{
-					var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidModifierAccess, Location.None, declarationSymbol.Name);
-					context.ReportDiagnostic(diag);
-					continue;
-				}
 
-				textStyleClassList.Add((declarationSymbol.Name, accessModifier, nameSpace, declarationSymbol.TypeArguments.GetGenericTypeArgumentsString(), declarationSymbol.GetGenericTypeConstraintsAsString()));
-			}
-		}
+		// Here we Collect all the Classes candidates from the first pipeline
+		// Then we merge then with the Maui.Controls that implements the desired interfaces
+		// Then we make sure they are unique and the user control doesn't inherit from any Maui control that implements the desired interface already
+		// Then we transform the ISymbol to be a type that we can compare and preserve the Incremental behavior of this Source Generator
+		var inputs = userGeneratedClassesProvider.Collect()
+			.Combine(mauiControlsAssemblySymbolProvider)
+			.SelectMany(static (x, _) => Deduplicate(x.Left!, x.Right).ToImmutableArray())
+			.Select(static (x, _) => TransformType(x));
 
-		var options = ((CSharpCompilation)compilation).SyntaxTrees[0].Options as CSharpParseOptions;
-		foreach (var textStyleClass in textStyleClassList)
-		{
-			var textColorToBuilder = @"
+		context.RegisterSourceOutput(inputs, Execution);
+	}
+
+	static void Execution(SourceProductionContext context, ModelToGenerate textStyleClass)
+	{
+
+		var textColorToBuilder = $@"
```

Comment:
We can change that in the future, this PR is just to fix the SG, so I didn't want to spend time converting it to string literals, if you see we concatenate the strings, this would add more work and more things to test. Feel free to open a discussion around that and if you want to submit a PR doing that (after we approve the discussion) will be awesome